src/app/core/services/auth/auth.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from "@angular/core";
+import { HttpClient } from "@angular/common/http";
+import { BehaviorSubject, Observable } from "rxjs";
+import { map } from "rxjs/operators";
+import { IUser } from "./../../../shared/interfaces/user";
+import { apiUrl } from "./../../../../environments/environment";
+
+@Injectable({
+  providedIn: "root"
+})
+export class AuthService {
+  private currentUserSubject: BehaviorSubject<IUser>;
+  public currentUser: Observable<IUser>;
+
+  constructor(private http: HttpClient) {
+    this.currentUserSubject = new BehaviorSubject<IUser>(
+      JSON.parse(localStorage.getItem("currentUser"))
+    );
+    this.currentUser = this.currentUserSubject.asObservable();
+  }
+
+  public get currentUserValue(): IUser {
+    return this.currentUserSubject.value;
+  }
+
+  register(user: IUser) {
+    return this.http.post<any>(`${apiUrl}/auth/register`, user)
```

Comment:
This logic better move to separate service, like 'data service' where will be going all server requests